--- SimPoint.3.2/analysiscode/CmdLineParser.cpp	2006-01-04 01:24:58.000000000 +0000
+++ SimPoint.3.2/analysiscode/CmdLineParser.cpp	2019-06-17 16:18:37.309236626 +0100
@@ -71,6 +71,7 @@


 #include "CmdLineParser.h"
+#include <string.h>

 bool CmdLineParser::parseCmdLine(int argc, char **argv) {
     for (int argNdx = 0; argNdx < argc; argNdx++) {
Only in SimPoint.3.2/analysiscode: CmdLineParser.d
Only in SimPoint.3.2/analysiscode: CmdLineParser.o
Only in SimPoint.3.2/analysiscode: Datapoint.d

--- SimPoint.3.2/analysiscode/Datapoint.h	2006-01-31 02:17:57.000000000 +0000
+++ SimPoint.3.2/analysiscode/Datapoint.h	2019-06-17 16:18:37.309236626 +0100
@@ -83,6 +83,7 @@

 #include <vector>
 #include <cstdio>
+#include <iostream>

 using namespace std;

Only in SimPoint.3.2/analysiscode: Datapoint.o
Only in SimPoint.3.2/analysiscode: Dataset.d

--- SimPoint.3.2/analysiscode/Dataset.h	2006-01-31 02:17:57.000000000 +0000
+++ SimPoint.3.2/analysiscode/Dataset.h	2019-06-17 16:18:37.313236565 +0100
@@ -83,6 +83,7 @@

 #include "Datapoint.h"
 #include <vector>
+#include <iostream>

 class Dataset : public vector<Datapoint> {
     public:
Only in SimPoint.3.2/analysiscode: Dataset.o

--- SimPoint.3.2/analysiscode/FVParser.cpp	2005-06-30 21:20:51.000000000 +0100
+++ SimPoint.3.2/analysiscode/FVParser.cpp	2019-06-17 16:18:37.313236565 +0100
@@ -79,6 +79,7 @@
 #include "FVParser.h"
 #include "Utilities.h"
 #include <string>
+#include <string.h>

 // take care of a difference between G++ 2.96 and 3.x
 #if (__GNUC__ >= 3)
Only in SimPoint.3.2/analysiscode: FVParser.d
Only in SimPoint.3.2/analysiscode: FVParser.o
Only in SimPoint.3.2/analysiscode: KMeans.d
Only in SimPoint.3.2/analysiscode: KMeans.o
Only in SimPoint.3.2/analysiscode: Logger.d
Only in SimPoint.3.2/analysiscode: Logger.o

--- SimPoint.3.2/analysiscode/Makefile	2005-06-30 22:55:04.000000000 +0100
+++ SimPoint.3.2/analysiscode/Makefile	2019-06-17 16:18:37.313236565 +0100
@@ -1,4 +1,5 @@
-CPPFLAGS = -Wall -pedantic -pedantic-errors -O3
+CPPFLAGS = -Wall -pedantic -pedantic-errors -O3 -std=c++11 -D_GLIBCXX_USE_CXX11_ABI=1
+# -D_GLIBCXX_USE_CXX11_ABI=1 has been added in order to support modern compiler version

 CXX = g++

Only in SimPoint.3.2/analysiscode: simpoint
Only in SimPoint.3.2/analysiscode: Simpoint.d
Only in SimPoint.3.2/analysiscode: Simpoint.o
Only in SimPoint.3.2/analysiscode: SimpointOptions.d
Only in SimPoint.3.2/analysiscode: SimpointOptions.o
Only in SimPoint.3.2/analysiscode: Utilities.d

--- SimPoint.3.2/analysiscode/Utilities.h	2006-02-01 19:39:48.000000000 +0000
+++ SimPoint.3.2/analysiscode/Utilities.h	2019-06-17 16:18:37.313236565 +0100
@@ -86,6 +86,9 @@
 #include "Dataset.h"
 #include "Logger.h"
 #include <string>
+#include <fstream>
+#include <stdlib.h>
+#include <limits.h>

 string toString(int i);
 string toString(double d);
@@ -131,6 +134,22 @@
                 exit(1);
             }
         }
+
+	static inline void check(std::ifstream& checkval, const string &msg){
+	    if(! checkval.is_open()){
+	    Logger::log() << "\nError: " << msg << endl;
+	    exit(1);
+	    }
+	}
+
+	static inline void check(std::ofstream& checkval, const string &msg){
+	    if(! checkval.is_open()){
+	    Logger::log() << "\nError: " << msg << endl;
+	    exit(1);
+	    }
+	}
+
+
 };


Only in SimPoint.3.2/analysiscode: Utilities.o
Only in SimPoint.3.2/bin: simpoint
